packages/plugin-relay/src/schema-builder.ts

```diff
@@ -521,7 +524,11 @@ schemaBuilderProto.connectionObject = function connectionObject(
                   this.options.relayOptions?.nodeFieldOptions?.nullable ??
                   false,
               },
-              resolve: (con) => (con.edges?.map((edge) => edge?.node) ?? []) as never,
+              resolve: (con) =>
+                completeValue(
+                  con.edges,
+                  (edges) => edges?.map((e) => e?.node) ?? (edgeListNullable ? null : []),
```

Comment:
@hayes not sure if this change needs mentioning, but previously `nodes` always `returned []` if edges was null, even if the field itsself was nullable.

Reply:
Just added a note